pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -29,6 +29,58 @@ This file is derived from NumPy 1.7. See NUMPY_LICENSE.txt
 #include <numpy/arrayscalars.h>
 #include <numpy/ndarraytypes.h>
 
+#if defined(_WIN32)
```

Comment:
FYI C23 has the `<stdckdint.h>` header which should replace these macros in a few years when that becomes widely available